sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala

```diff
@@ -130,7 +130,10 @@ private[spark] object HiveUtils extends Logging {
 
   val CONVERT_METASTORE_ORC = buildConf("spark.sql.hive.convertMetastoreOrc")
     .doc("When set to true, the built-in ORC reader and writer are used to process " +
-      "ORC tables created by using the HiveQL syntax, instead of Hive serde.")
+      "ORC tables created by using the HiveQL syntax, instead of Hive serde." +
+      "Enabling this parameter will cause insert overwrite when reading the table," +
+      "and the file format of this table is ORC This error will be thrown[], " +
```

Comment:
@dh20, which case are you trying to elabourate?

Reply:
@HyukjinKwon When I query an orc format hive table and overwrite the query result into this orc table, sparksql will throw this error -> [org.apache.spark.sql.AnalysisException: Cannot overwrite a path that is also being read from], when I set this parameter to false, Hql will execute normally

Reply:
I think it's weird to overwrite a path is being read from. Is it something officially supported from Hive?

Reply:
@HyukjinKwon Yes, Hive officially supported this function. This SQL was originally run on hive, and I switched sparksql before this problem appeared

Reply:
I think it's fine enough to say we use Spark's when `spark.sql.hive.convertMetastoreOrc` is enabled. Otherwise, we will have to list up every behaviour differences between Spark and Hive in this configuration doc.

Reply:
@HyukjinKwon Yes, I think so, but now the official explanation of spark for this parameter is aimed at the difference between spark Orc vector readers, so I think its explanation should be improved

Reply:
The doc already says that "the built-in ORC reader" vs "Hive serde". I think it's fine.

Reply:
@dh20 . It seems that you want to describe one of the differences between Spark data source tables and Hive serde tables. May I ask a few questions?

- Is the situation different from `Parquet`? Parquet also has the same conf `spark.sql.hive.convertMetastoreParquet`.

- Is the added statement correct always? Specifically, the following statement looks wrong in some situations. For example, did you try to use `PartitionOverwriteMode.DYNAMIC`?
> When I query an orc format hive table and overwrite the query result into this orc table, sparksql will throw this error -> [org.apache.spark.sql.AnalysisException: Cannot overwrite a path that is also being read from], when I set this parameter to false, Hql will execute normally

- Which Apache Spark version are you using? The situation will be changed according to the Spark versions. For example,
  - SPARK-30112 implemented `Allow insert overwrite same table if using dynamic partition overwrite` at Apache Spark 3.0.0. 
  - SPARK-33887 is also trying to implement `Allow insert overwrite same table with static partition if using dynamic partition overwrite mode` at Apache Spark 3.2.0.

Reply:
@dongjoon-hyun Hello, I'll answer your questions one by one.

- 1.When the format of the table is parquet, there is no problem to insert and overlay the same table, no matter whether this parameter is added or not【 spark.sql.hive .convertMetastoreParquet=false】

- 2.[PartitionOverwriteMode.DYNAMIC] It can't solve the problem

- 3.I use spark 3.0.0, spark-30112 does not solve this problem.


.

Reply:
Could you file a JIRA for this with your specific procedure? This one is a valid bug.
> When the format of the table is parquet, there is no problem to insert and overlay the same table, no matter whether this parameter is added or not【 spark.sql.hive .convertMetastoreParquet=false】

For this one, I think your statement in this PR is too broader because we have a unit test coverage with Dynamic which is added by SPARK-30112. When you say `this problem`, you should be specific to avoid ambiguity.
> [PartitionOverwriteMode.DYNAMIC] It can't solve the problem
> I use spark 3.0.0, spark-30112 does not solve this problem.